zebra-state/src/service/finalized_state/zebra_db.rs

```diff
@@ -307,78 +323,92 @@ impl FinalizedState {
-        current_value_pool: ValueBalance<NonNegative>,
-    ) -> Result<DiskWriteBatch, BoxError> {
+        all_utxos_spent_by_block: HashMap<transparent::OutPoint, transparent::Utxo>,
+        // TODO: make an argument struct for all the current note commitment trees & history
+        mut note_commitment_trees: NoteCommitmentTrees,
+        history_tree: HistoryTree,
+        value_pool: ValueBalance<NonNegative>,
+    ) -> Result<(), BoxError> {
         let hash_by_height = db.cf_handle("hash_by_height").unwrap();
         let height_by_hash = db.cf_handle("height_by_hash").unwrap();
         let block_by_height = db.cf_handle("block_by_height").unwrap();
-        let tx_by_hash = db.cf_handle("tx_by_hash").unwrap();
-        let utxo_by_outpoint = db.cf_handle("utxo_by_outpoint").unwrap();
-
-        let sprout_nullifiers = db.cf_handle("sprout_nullifiers").unwrap();
-        let sapling_nullifiers = db.cf_handle("sapling_nullifiers").unwrap();
-        let orchard_nullifiers = db.cf_handle("orchard_nullifiers").unwrap();
-
-        let sprout_anchors = db.cf_handle("sprout_anchors").unwrap();
-        let sapling_anchors = db.cf_handle("sapling_anchors").unwrap();
-        let orchard_anchors = db.cf_handle("orchard_anchors").unwrap();
-
-        let sprout_note_commitment_tree_cf = db.cf_handle("sprout_note_commitment_tree").unwrap();
-        let sapling_note_commitment_tree_cf = db.cf_handle("sapling_note_commitment_tree").unwrap();
-        let orchard_note_commitment_tree_cf = db.cf_handle("orchard_note_commitment_tree").unwrap();
-        let history_tree_cf = db.cf_handle("history_tree").unwrap();
-
-        let tip_chain_value_pool = db.cf_handle("tip_chain_value_pool").unwrap();
 
         let FinalizedBlock {
             block,
             hash,
             height,
-            new_outputs,
-            transaction_hashes,
-        } = finalized;
+            ..
+        } = &finalized;
 
         // The block has passed contextual validation, so update the metrics
-        FinalizedState::block_precommit_metrics(&block, hash, height);
+        FinalizedState::block_precommit_metrics(block, *hash, *height);
 
         // Index the block
         self.zs_insert(hash_by_height, height, hash);
         self.zs_insert(height_by_hash, hash, height);
-        self.zs_insert(block_by_height, height, &block);
+        self.zs_insert(block_by_height, height, block);
 
         // # Consensus
         //
         // > A transaction MUST NOT spend an output of the genesis block coinbase transaction.
         // > (There is one such zero-valued output, on each of Testnet and Mainnet.)
         //
         // https://zips.z.cash/protocol/protocol.pdf#txnconsensus
+        //
+        // By returning early, Zebra commits the genesis block and transaction data,
+        // but it ignores the genesis UTXO and value pool updates.
+        //
+        // TODO: commit transaction data but not UTXOs in the next PR.
```

Comment:
```suggestion
        // TODO: commit transaction data but not UTXOs.
```
I think we should not say "next PR" in any comment.

Reply:
This comment gets removed in the next PR.

Reply:
I know but i think we should avoid them anyway.

Reply:
I'll try not to include them in any other PRs.